src/hydra-queue-runner/queue-monitor.cc

```diff
@@ -74,6 +75,53 @@ struct PreviousFailure : public std::exception {
 };
 
 
+bool State::getQueuedBuildsInEval(Connection & conn,
+    ref<Store> destStore, unsigned int evalId)
```

Comment:
There is a lot of code duplication between this and `getQueuedBuilds`, maybe that can be avoided?